src/Physics/HadronTransport/Intranuke2025.cxx

```diff
@@ -0,0 +1,406 @@
+#include "Framework/Conventions/GBuild.h"
+#include "Framework/Conventions/Constants.h"
+#include "Framework/Conventions/Controls.h"
+#include "Framework/GHEP/GHepStatus.h"
+#include "Framework/GHEP/GHepRecord.h"
+#include "Framework/GHEP/GHepParticle.h"
+#include "Physics/HadronTransport/Intranuke2025.h"
+#include "Physics/HadronTransport/INukeHadroData2025.h"
+#include "Physics/HadronTransport/INukeHadroFates.h"
+#include "Physics/HadronTransport/INukeMode.h"
+#include "Physics/HadronTransport/INukeUtils2025.h"
+#include "Framework/Interaction/Interaction.h"
+#include "Framework/Messenger/Messenger.h"
+#include "Framework/Numerical/RandomGen.h"
+#include "Framework/Numerical/Spline.h"
+#include "Framework/ParticleData/PDGLibrary.h"
+#include "Framework/ParticleData/PDGCodes.h"
+#include "Framework/ParticleData/PDGCodeList.h"
+#include "Framework/ParticleData/PDGUtils.h"
+#include "Framework/Utils/PrintUtils.h"
+#include "Physics/NuclearState/NuclearUtils.h"
+
+using std::ostringstream;
+
+using namespace genie;
+using namespace genie::utils;
+using namespace genie::constants;
+using namespace genie::controls;
+
+//___________________________________________________________________________
+Intranuke2025::Intranuke2025() :
+EventRecordVisitorI()
+{
+
+}
+//___________________________________________________________________________
+Intranuke2025::Intranuke2025(string name) :
+EventRecordVisitorI(name)
+{
+
+}
+//___________________________________________________________________________
+Intranuke2025::Intranuke2025(string name, string config) :
+EventRecordVisitorI(name, config)
+{
+
+}
+//___________________________________________________________________________
+Intranuke2025::~Intranuke2025()
+{
+
+}
+//___________________________________________________________________________
+void Intranuke2025::ProcessEventRecord(GHepRecord * evrec) const
+{
+  // Do not continue if there is no nuclear target
+  GHepParticle * nucltgt = evrec->TargetNucleus();
+  if (!nucltgt) {
+    LOG("HNIntranuke2025", pINFO) << "No nuclear target found - INTRANUKE exits";
+    return;
+  }
+
+  // Decide tracking radius for the current nucleus (few * R0 * A^1/3)
+  this->SetTrackingRadius(nucltgt);
+
+  // Understand what the event generation mode is (hadron/photon-nucleus,
+  // lepton-nucleus, nucleon decay) from the input event.
+  // The determined mode has an effect on INTRANUKE behaviour (how to lookup
+  // the residual nucleus, whether to set an intranuclear vtx etc) but it
+  // does not affect the INTRANUKE physics.
+  fGMode = evrec->EventGenerationMode();
+
+  // For lepton-nucleus scattering and for nucleon decay intranuclear vtx
+  // position (in the target nucleus coord system) is set elsewhere.
+  // This method only takes effect in hadron/photon-nucleus interactions.
+  // In this special mode, an interaction vertex is set at the periphery
+  // of the target nucleus.
+  if(fGMode == kGMdHadronNucleus ||
```

Comment:
How to set vertex get set if it isn't one of these two modes?  Perhaps mention that, otherwise users might expect this to have an `else` clause.

Also, if it's not otherwise restricted by inheritance would it be possible to change the function name to be more descriptive, ala `GenerateVertexPeriphery()` or something.